sonnydashboard/server.js

```diff
@@ -0,0 +1,99 @@
+var   express     = require('express'),
+      bodyParser  = require('body-parser'),
+      path        = require('path'),
+      io          = require('socket.io'),
+      mongoose    = require('mongoose')
+      app         = express();
+
+app.use(bodyParser.urlencoded({extended: true}));
+
+app.set('view engine', 'ejs');
+app.set('views', path.join(__dirname, './views'));
+
+//Routes
+
+app.get('/', function(req, res) {
+  Animal.find({}, function(err, animals) {
+    if(err) {
+      console.log('something went wrong');
+    } else {
+        res.render('index', {animals: animals});
+    }
+  })
+})
+
+app.get('/species/new', function(req, res) {
+  res.render('create');
+})
+
+app.post('/species', function(req, res) {
+  var animal = new Animal({
+    Animal: req.body.Animal
+  });
+  animal.save(function(err) {
+    if(err) {
+      console.log('something went wrong');
+    } else {
+      console.log('successfully added animal');
+      res.redirect('/');
+    }
+  })
+})
+
+app.get('/species/:id', function(req, res) {
+  Animal.findOne({_id: req.param.id}, function(err, animals) {
+    if(err) {
+      console.log('something went wrong');
+    } else {
+      console.log(animals);
+      res.render('show', {animals: req.body.Animal});
```

Comment:
Ideally we don't want to bounce data from view to view. req.body.Animal came from the view, and then we immediately send it back into another view. Since you're taking a trip to the database anyway, I recommend passing the `animals` that `Animal.findOne` returns.